src/redux/reducers/accommodationReducer.js

```diff
@@ -0,0 +1,38 @@
+import {
+    ADD_ACCOMMODATION_REQUEST,
+    VIEW_ALL_ACCOMMODATIONS, 
+    VIEW_ALL_ACCOMMODATIONS_SUCCESS,
+    VIEW_ALL_ACCOMMODATIONS_FAILURE
+  } from '../types/accommodationTypes';
+  
+  const initialState = {
+    loading: false,
+    allAccommodations:[],
+    error:''
+  };
+  const accommodationReducer = (state = initialState, action) => {
+      switch (action.type) {
+        case VIEW_ALL_ACCOMMODATIONS:
+          return {
+            ...state,
+            loading: true
+        }
+        case VIEW_ALL_ACCOMMODATIONS_SUCCESS:
+          return {
+            loading: false,
+            allAccommodations: action.payload
+        }
+        case VIEW_ALL_ACCOMMODATIONS_FAILURE:
+          return {
+            loading: false,
+            error: action.payload
+        }
+        case ADD_ACCOMMODATION_REQUEST:
+          return initialState;
+          default:
+           return state;
+      }
```

Comment:
@hakoemmy what is the purpose of this code?

Reply:
@AbdoulNuru SInce, I combined your reducer into mine, I had to bring the switch case there in ```src/redux/reducers/accommodationReducer.js```  but found that you were returning an empty array which was interfering with the key ```accommodation``` that uses the store as ```viewAllAccommodations```